pkg/controllers/deprovisioning/consolidation.go

```diff
@@ -93,17 +90,17 @@ func (c *consolidation) markConsolidated() {
 }
 
 // ShouldDeprovision is a predicate used to filter deprovisionable nodes
-func (c *consolidation) ShouldDeprovision(_ context.Context, cn *Candidate) bool {
-	if val, ok := cn.Annotations()[v1alpha5.DoNotConsolidateNodeAnnotationKey]; ok {
-		c.recorder.Publish(deprovisioningevents.Unconsolidatable(cn.Node, cn.Machine, fmt.Sprintf("%s annotation exists", v1alpha5.DoNotConsolidateNodeAnnotationKey))...)
-		return val != "true"
+func (c *consolidation) shouldDeprovision(_ context.Context, cn *Candidate, isEmptyConsolidation bool) bool {
+	if cn.Annotations()[v1alpha5.DoNotConsolidateNodeAnnotationKey] == "true" {
+		c.recorder.Publish(deprovisioningevents.Unconsolidatable(cn.Node, cn.NodeClaim, fmt.Sprintf("%s annotation exists", v1alpha5.DoNotConsolidateNodeAnnotationKey))...)
+		return false
 	}
-	if cn.provisioner == nil {
-		c.recorder.Publish(deprovisioningevents.Unconsolidatable(cn.Node, cn.Machine, "provisioner is unknown")...)
+	if cn.nodePool.Spec.Deprovisioning.ConsolidationPolicy == v1beta1.ConsolidationPolicyNever {
+		c.recorder.Publish(deprovisioningevents.Unconsolidatable(cn.Node, cn.NodeClaim, fmt.Sprintf("nodepool %s has all consolidation disabled by consolidation policy: %s", cn.nodePool.Name, cn.nodePool.Spec.Deprovisioning.ConsolidationPolicy))...)
```

Comment:
nit: I feel like you probably just need the name here of the nodepool. IIUC, this will just read 
"node pool default has all consolidation disabled by consolidation policy: never", which I feel is a little redundant. Up to you though.